app/controllers/ReturnObligationsController.scala

```diff
@@ -35,11 +35,15 @@ class ReturnObligationsController @Inject()(val messagesApi: MessagesApi,
                                             implicit val appConfig: AppConfig)
   extends AuthorisedController {
 
-  def completedReturns(): Action[AnyContent] = authorisedAction { implicit request =>
+  def completedReturns(year: Int): Action[AnyContent] = authorisedAction { implicit request =>
     implicit user =>
-      for {
-        returnObligations <- handleReturnObligations(user)
-      } yield Ok(views.html.returns.completedReturns(returnObligations))
+      if(validateSearchYear(year)) {
+        for {
+          returnObligations <- handleReturnObligations(user, year)
+        } yield Ok(views.html.returns.completedReturns(returnObligations))
```

Comment:
I still think this looks cleaner replaced by: 
```
        handleReturnObligations(user, year).map { obligations =>
          Ok(views.html.returns.completedReturns(obligations))
        }
```
instead of for yield when there is only one future involved